File: TheaterSuggestions/Java/SeatingPlaceSuggestionsAcceptanceTests/src/test/java/org/weaveit/seatssuggestionsacceptancetests/Row.java
package org.weaveit.seatssuggestionsacceptancetests;

import java.util.ArrayList;
import java.util.List;

public class Row {
    private String name;
    private List<SeatingPlace> seatingPlace = new ArrayList<>();

    public Row(String name, List<SeatingPlace> seatingPlace) {
        this.name = name;
        this.seatingPlace = seatingPlace;
    }

    public List<SeatingPlace> seatingPlaces() {
        return seatingPlace;
    }
}

File: TheaterSuggestions/CSharp/SeatingPlaceSuggestions/Row.cs
﻿namespace SeatsSuggestions;

public class Row(string name, List<SeatingPlace> seats)
{
    public string Name { get; } = name;
    public List<SeatingPlace> SeatingPlaces { get; } = seats;

    public SeatingOptionIsSuggested SuggestSeatingOption(int partyRequested, PricingCategory pricingCategory)
    { 
        var seatAllocation = new SeatingOptionIsSuggested(partyRequested, pricingCategory);
        
        foreach (var seat in SeatingPlaces)
        {
            if (seat.IsAvailable() && seat.MatchCategory(pricingCategory))
            {
                seatAllocation.AddSeat(seat);
                if (seatAllocation.MatchExpectation()) return seatAllocation;
            }
        }
        return new SeatingOptionIsNotAvailable(partyRequested, pricingCategory);
    }

    public void AddSeat(SeatingPlace seatingPlace)
    {
        SeatingPlaces.Add(seatingPlace);
    }
}